### PR TITLE
Fs cloning handler

### PR DESCRIPTION
**Pull request by mpacer**
This adds a GET handler for file-system cloning.

Requires #157.

### CODE REVIEW

File: bookstore/clone.py [hunk 1]
```diff
@@ -339,3 +339,77 @@ def validate_relpath(relpath, settings, log):
...
+    `Jupyter Notebook reference on Custom Handlers <https://jupyter-notebook.readthedocs.io/en/stable/extending/handlers.html#registering-custom-handlers>`_
+    """
+
+    def initialize(self):
+        """Helper to retrieve bookstore setting for the session."""
+        self.bookstore_settings = BookstoreSettings(config=self.config)
+
+    @web.authenticated
+    async def get(self):
+        """GET /bookstore/fs-clone?s3_bucket=<your_s3_bucket>&s3_key=<your_s3_key>
```
Comment (chyzzqo2) [inline bookstore/clone.py]: update?

File: bookstore/tests/test_clone.py [hunk 2]
```diff
@@ -301,3 +302,89 @@ def test_validate_relpath_escape_basedir(caplog):
...
+
+    @gen_test
+    async def test_get_no_param(self):
+        empty_handler = self.get_handler('/bookstore/fs-clone')
+        with pytest.raises(HTTPError):
+            await empty_handler.get()
+
+    @gen_test
+    async def test_get_empty_relpath(self):
+        no_bucket_handler = self.get_handler('/bookstore/fs-clone?relpath=')
```
Comment (chyzzqo2) [inline bookstore/tests/test_clone.py]: name seems misleading

File: bookstore/tests/test_clone.py [hunk 3]
```diff
@@ -301,3 +302,89 @@ def test_validate_relpath_escape_basedir(caplog):
...
+        with pytest.raises(HTTPError):
+            await no_bucket_handler.get()
+
+    @gen_test
+    async def test_get_escape_basedir(self):
+        no_bucket_handler = self.get_handler('/bookstore/fs-clone?relpath=../hi')
```
Comment (chyzzqo2) [inline bookstore/tests/test_clone.py]: name seems misleading

File: bookstore/tests/test_clone.py (hunk 2 above)
Comment (mpacer) [reply bookstore/tests/test_clone.py]: good point!

File: bookstore/tests/test_clone.py (hunk 3 above)
Comment (mpacer) [reply bookstore/tests/test_clone.py]: The perils of copying other code as a template 😛

**Pull request merged**